pkg/arvo/sys/vane/eyre.hoon

```diff
@@ -2900,6 +3618,92 @@
     ?:  =(*@da next)  expiry-time
     (min next expiry-time)
   ::
+  ++  eauth
+    =*  auth  auth.server-state.ax
+    =*  args  [[eny duct now rof] server-state.ax]
+    ^-  [(list move) _http-server-gate]
+    ~|  [wire +<.sign]
+    ?+  t.wire  !!
+        [%plea @ ~]
+      =/  =ship  (slav %p i.t.t.wire)
+      ::
+      ?:  |(?=(^ dud) ?=([%ames %lost *] sign))
+        %-  %+  trace:(per-server-event args)  0
+            ?~  dud  |.("eauth: lost boon from {(scow %p ship)}")
+            |.("eauth: crashed on %{(trip +<.sign)} from {(scow %p ship)}")
+        ::NOTE  when failing on pending attempts, we just wait for the timer
+        ::      to clean up. when failing on live sessions, well, we should
+        ::      just be careful not to crash when receiving %shut boons.
+        ::      (we do not want to have the nonce in the wire, so this is the
+        ::      best handling we can do. the alternative is tracking)
+        [~ http-server-gate]
+      ::
+      ?:  ?=([%ames %done *] sign)
+        ?~  error.sign
+          %-  %+  trace:(per-server-event args)  2
+            |.("eauth: ack from {(scow %p ship)}")
+          [~ http-server-gate]
+        ::NOTE  we would *really* like to be calling on-fail:client here, but
+        ::      we can't, because we don't know the nonce. we *can* say:
+        ::      - if %open was nacked, we will just time out the attempt.
+        ::        - but! we could be showing the user an error sooner...
+        ::      - if %shut was nacked, we optimistically deleted it locally,
+        ::        and the remote session will eventually expire, assuming it
+        ::        goes unused.
+        ::        - but! this may not be the case if we were %shutting because
+        ::          our device/session/cookie was compromised...
+        ::          - but! good hosts shouldn't crash on %shuts at all...
```

Comment:
The way to fix this would be to keep a per-ship `qeu` for outstanding plea's. This would let us correlate the nack to a nonce and eagerly send an error response.

The only "normal" case in which we'd get a nack here is if the server has no eauth-url (no one has ever logged into it) and the eauth-flow was begun on the client ship. This doesn't seem like a very priority, but it would be hard to add later, so it'd be best to track the state now.